devito/core/autotuning.py

```diff
@@ -75,16 +75,13 @@ def autotune(operator, args, level, mode):
     # Detect the time-stepping Iteration; shrink its iteration range so that
     # each autotuning run only takes a few iterations
     steppers = {i for i in flatten(trees) if i.dim.is_Time}
-    if len(steppers) == 0:
-        stepper = None
-        timesteps = 1
-    elif len(steppers) == 1:
+    if len(steppers) == 1:
```

Comment:
if CI doesn't fail, either test suite is flawed or we somewhat changed things over time such that it's never the case.

Now, the second case to me is unlikely...one could definitely try auto-tuning without a time loop. Hence, the test suite deserves an update !

Reply:
The 0 case is removed since devito does not apply loop blocking for non time-iterative computations. 
https://github.com/devitocodes/devito/blob/45c75a7591e8f098c8b48c21ea639d5bf1bf368a/devito/passes/clusters/blocking.py#L148

CI passes.

Idea: blockrelax should allow blocking non-time-iterative loops, right? It skips the heuristics. Then we need to add some tests for this.

i.e. block relax on matrix multiplication example?
I need to check

Reply:
sounds good to me yeah